funding/commitment_type_negotiation.go

```diff
@@ -30,6 +30,15 @@ func negotiateCommitmentType(channelType *lnwire.ChannelType,
 				*channelType, local, remote,
 			)
 		}
+
+		// If we're the funder, and we are attempting to use an
+		// explicit channel type, but the remote party doesn't signal
+		// the bit, then we actually want to exit here, to ensure the
+		// user doesn't end up with an unexpected channel type via
+		// implicit negotiation.
+		if isFunder {
```

Comment:
Does this mean we gonna fail when opening a channel with an older version of lnd when using the explicit channel type?

Reply:
Yes, which is what we want, in order to prevent a user that wants chan type `T` ending up with type `K`.

Reply:
With the old logic, we'd exit similarly here: https://github.com/guggero/lnd/blob/e7885f2bde37e9d2ab8c496fc66b9e910e7e5d49/funding/commitment_type_negotiation.go#L33

Reply:
Another option is to just pick a type for the user here: https://github.com/lightningnetwork/lnd/blob/master/rpcserver.go#L1995

Then we always set a channel type, regardless of if the bit is signaled or not

Reply:
If we always set a bit, and don't allow fallback, we won't be able to open to v0.13.3 lnd nodes or non-lnd nodes w/o the explicit chan type feature bit (excluding prior ones).

Reply:
> If we always set a bit, and don't allow fallback, we won't be able to open to v0.13.3 lnd nodes or non-lnd nodes w/o the explicit chan type feature bit (excluding prior ones).

Ah yeah that's correct, which is why we only use this behavior when the user explicitly specifies a type.